validation/pidfile.go

```diff
@@ -0,0 +1,63 @@
+package main
+
+import (
+	"errors"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	tap "github.com/mndrix/tap-go"
+	"github.com/opencontainers/runtime-tools/validation/util"
+)
+
+func main() {
+	t := tap.New()
+	t.Header(0)
+
+	tempDir, err := ioutil.TempDir("", "oci-pid")
+	if err != nil {
+		util.Fatal(err)
+	}
+	defer os.RemoveAll(tempDir)
+	tempPidFile := filepath.Join(tempDir, "pidfile")
+
+	config := util.LifecycleConfig{
+		Actions: util.LifecycleActionCreate | util.LifecycleActionDelete,
+		PreCreate: func(r *util.Runtime) error {
+			r.PidFile = tempPidFile
+			return nil
+		},
+		PostCreate: func(r *util.Runtime) error {
+			pidData, err := ioutil.ReadFile(tempPidFile)
+			if err != nil {
+				return err
+			}
+			pid, err := strconv.Atoi(string(pidData))
+			if err != nil {
+				return err
+			}
+			state, err := r.State()
+			if err != nil {
+				return err
+			}
+			if state.Pid != pid {
+				return errors.New("Wrong pid in the pidfile")
+			}
+			return nil
+		},
+		PreDelete: func(r *util.Runtime) error {
+			return util.WaitingForStatus(*r, util.LifecycleStatusCreated, time.Second*10, time.Second*1)
```

Comment:
We shouldn't need this, because a container [must be created before `create` returns][1].  Maybe a runc bug?

[1]: https://github.com/opencontainers/runtime-spec/blame/v1.0.1/runtime.md#L20

Reply:
`runc` does not implement ['creating' lifecycle ](https://github.com/opencontainers/runc/blob/master/libcontainer/container.go#L17), 
it is always 'created' or 'fails'.
From the spec, it should be like this:
1. user run `runX create x` 
2. user run `runX status x`
    it should be 'creating' (assuming the creating takes a long time).

Reply:
> `runc` does not implement 'creating' lifecycle...

That's definitely a bug.  And there's also some [CLI spec wording][1] about `create`'s exit timing.  I think we can drop this, and I'll file a runc patch.

[1]: https://github.com/opencontainers/runtime-tools/blame/7ac117af23fb7465142f16ac0d22277c9c82fefa/docs/command-line-interface.md#L79